src/screens/OtherProfile/index.js

```diff
@@ -164,91 +133,14 @@ const OtherProfile = () => {
   }, []);
 
   React.useEffect(() => {
-    netInfo.addEventListener((state) => {
-      if (state.isConnected) {
-        initData();
-      } else {
-        handleOfflineMode();
-      }
-    });
     initData();
   }, []);
 
-  React.useEffect(() => {
-    getOtherFeeds();
-  }, [other_id]);
-
   const initData = () => {
-    setInitLoading(true);
     setUserId(params.data.user_id);
     setUsername(params.data.username);
-    fetchOtherProfile();
-  };
-
-  const checkUserBlockHandle = async (userId, callback) => {
-    const status = await netInfo.fetch();
-    if (status.isConnected) {
-      try {
-        const sendData = {
-          user_id: userId
-        };
-        const processGetBlock = await checkUserBlock(sendData);
-        if (callback) callback();
-        if (processGetBlock.status === 200) {
-          setBlockStatus(processGetBlock.data.data);
-        }
-      } catch (e) {
-        if (__DEV__) {
-          console.log(e, 'error');
-        }
-      }
-    }
-  };
-
-  const handleSaveDataOtherProfile = (data) => {
-    setDataMain(data);
-    setDataMainBio(data.bio);
-    checkUserBlockHandle(data.user_id);
-    setOtherId(data.user_id);
-    setLoading(false);
-    setInitLoading(false);
-  };
-
-  const fetchOtherProfile = async () => {
-    const status = await netInfo.fetch();
-    if (status.isConnected) {
-      try {
-        handleOfflineMode();
-        const result = await getOtherProfile(params?.data?.username);
-        if (result.code === 200) {
-          handleSaveDataOtherProfile(result.data);
-        }
-        StorageUtils.otherProfileData.setForKey(
-          params?.data?.username,
-          JSON.stringify(result.data)
-        );
-      } catch (e) {
-        if (e.response && e.response.data && e.response.data.message) {
-          SimpleToast.show(e.response.data.message, SimpleToast.SHORT);
-        }
-        setLoading(false);
-        setInitLoading(false);
-      }
-    } else {
-      handleOfflineMode();
-    }
   };
 
-  const handleOfflineMode = async () => {
-    const cache = await StorageUtils.otherProfileData.getForKey(params?.data?.username);
-    if (cache) {
-      const data = JSON.parse(cache);
-      handleSaveDataOtherProfile(data);
-    } else {
-      setLoading(false);
-      setInitLoading(false);
-    }
-  };
   const onShare = async () => ShareUtils.shareUserLink(username);
 
   const handleSetUnFollow = async () => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initData` function is called inside a `useEffect` hook without any dependencies but it is defined inside another `useEffect` hook with an empty dependency array. This could potentially lead to a race condition where `initData` might be called before it is defined. To avoid this, you should define `initData` outside of the `useEffect` hooks or ensure that it is defined before it is called.

```diff
-  React.useEffect(() => {
-    initData();
-  }, []);
-
-  const initData = () => {
-    setUserId(params.data.user_id);
-    setUsername(params.data.username);
-  };
+  const initData = useCallback(() => {
+    setUserId(params.data.user_id);
+    setUsername(params.data.username);
+  }, [params.data]);
+
+  React.useEffect(() => {
+    initData();
+  }, [initData]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->